components/bookmarksList/index.jsx

```diff
@@ -0,0 +1,175 @@
+import ResourceLink from "../resourceLink";
+import SortedTableCarat from "../sortedTableCarat";
+import Spinner from "../spinner";
+import styles from "./styles";
+import { RECALLS_PROPERTY_IRI } from "../../src/solidClientHelpers/bookmarks";
+import useFindResourceOwner from "../../src/hooks/useFindResourceOwner";
+
+const useStyles = makeStyles((theme) => createStyles(styles(theme)));
+
+function OwnerDisplayName({ iri }) {
+  const { ownerName } = useFindResourceOwner(iri);
+  const [loading, setLoading] = useState(true);
+  useEffect(() => {
+    if (typeof ownerName === "undefined") {
+      setLoading(true);
+    } else {
+      setLoading(false);
+    }
+  }, [ownerName, setLoading]);
+
+  if (loading) {
+    return <Skeleton width={150} />;
+  }
+  return <span>{ownerName}</span>;
+}
+
+const bookmarkBody = ({ value }) => {
+  return <Bookmark iri={value} />;
+};
+
+const titleBody = ({ row, value, className }) => {
+  const bookmarkedResourceIri = row.original.col0;
+  return (
+    <ResourceLink
+      containerIri={bookmarkedResourceIri}
+      resourceIri={bookmarkedResourceIri}
+      className={className}
+    >
+      {value}
+    </ResourceLink>
+  );
+};
+
+const ownerBody = ({ value }) => {
+  if (!value) {
+    return null;
+  }
+  return <OwnerDisplayName iri={value} />;
+};
+
+function BookmarksList() {
+  const { bookmarks } = useContext(BookmarksContext);
+  const tableClass = PrismTable.useTableClass("table", "inherits");
+  const classes = useStyles();
+  const bem = useBem(classes);
+  const tableLinkClassName = bem("table__link");
+  const [search, setSearch] = useState();
+
+  const isLoading = !bookmarks;
+
+  if (isLoading) return <Spinner />;
+  const { dataset } = bookmarks;
+  const bookmarksList = getThingAll(dataset).map((b) => {
+    return {
+      thing: b,
+      dataset,
+    };
+  });
+
+  const ascIndicator = (
+    <>
+      {` `}
+      <SortedTableCarat sorted />
+    </>
+  );
+  const descIndicator = (
+    <>
+      {" "}
+      <SortedTableCarat sorted sortedDesc />
+    </>
+  );
+
+  return (
+    <>
+      <PageHeader title="Bookmarks">
+        <TextField
+          margin="dense"
+          variant="outlined"
+          onChange={(e) => setSearch(e.target.value)}
+        />
+      </PageHeader>
+      <Container>
+        <Table
+          things={bookmarksList}
+          className={clsx(tableClass, bem("table"))}
+          filter={search}
+          ascIndicator={ascIndicator}
+          descIndicator={descIndicator}
+        >
+          <TableColumn
+            property={RECALLS_PROPERTY_IRI}
+            dataType="url"
+            header=""
+            body={bookmarkBody}
+          />
+          <TableColumn
+            property={dct.title}
+            header="Name"
+            className={tableLinkClassName}
+            // eslint-disable-next-line prettier/prettier
+            body={({ row, value }) => titleBody({ row, value, className: tableLinkClassName })}
```

Comment:
was this line not causing you the same re-render issues the others were?

Reply:
With the ownerBody not re-rendering, each time we delete a bookmark we don't need to re-fetch the owner name so that fixed most of it.